high-level.lisp

```diff
@@ -888,3 +888,35 @@ with upper left block with dimension P-by-Q. Returns the intermediate representa
     (flet ((calc-i-j (i j) (* (ref a (floor i mb) (floor j nb))
                               (ref b (mod i mb) (mod j nb)))))
       (reduce #'kron rest :initial-value (tabulate (* ma mb) (* na nb) #'calc-i-j)))))
+
+(defun num-to-bits (num)
+  (loop for i below (integer-length num)
+	collect (if (logbitp i num) 1 0)))
+
+(defun bin-decomp (num)
+  (let ((bits (num-to-bits num)))
+    (loop for i below (length bits)
+          for b in bits
+          when (> b 0)
+          collect (* b (expt 2 i)))))
+
+(defun exptm (m power)
+  "Compute the matrix M raised to power POWER."
+  (assert (= (matrix-rows m) (matrix-cols m)) () "M is not a square matrix")
+  ;; TODO Permit real-valued powers via #'EXPM
+  (assert (integerp power) () "POWER is not an integer")
```

Comment:
use `(check-type power integer)`

Reply:
Actually I would just check that it's a fixnum. You're not going to be doing bignum matrix powers.

```
(check-type power fixnum)
```